mne/channels/channels.py

```diff
@@ -228,6 +230,70 @@ def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
         return _get_channel_types(self.info, picks=picks, unique=unique,
                                   only_data_chs=only_data_chs)
 
+    def get_ch_positions(self, picks=None):
+        """Get a dictionary of channel positions for each channel.
+
+        Channel positions are in meters in the specified coordinate frame
+        of the montage.
+
+        Parameters
+        ----------
+        picks : str | list | slice | None
+            None gets good data indices.
+
+        Returns
+        -------
+        ch_positions : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding list of xyz coordinates in the montage
+             coordinate frame.
+        coord_frame : str
+            The coordinate frame of the montage's xyz coordinates.
+        """
+        # get the channel positions
+        picks = _picks_to_idx(self.info, picks)
+        chs = self.info['chs']
+        ch_names = self.info['ch_names']
+        pos = np.array([chs[k]['loc'][:3] for k in picks])
+
+        # extract landmark coords and coordinate frame
+        _, coord_frame_int = _get_fid_coords(self.info['dig'])
```

Comment:
you should use the coordframe provided in chs structure. Don't use dig here.

Reply:
I see, I then can create a dictionary of `{ch_name: ch_coord_frame}` then? That is what I return now instead of a single coord frame.